pkg/operator/controller/ingress/deployment.go

```diff
@@ -1436,3 +1456,29 @@ func clipHAProxyTimeoutValue(val string) (string, error) {
 	}
 	return val, nil
 }
+
+// durationToHAProxyTimespec converts a time.Duration into a number that
+// HAProxy can consume, in the simplest unit possible. If the value would be
+// truncated by being converted to seconds, it outputs in milliseconds,
+// otherwise if the value wouldn't be truncated by converting to seconds, but
+// would if converted to minutes, it outputs in seconds, etc. up to a maximum
+// unit in hours (the largest unit natively supported by time.Duration).
+//
+// Also truncates values to the maximum length HAProxy allows if the value is
+// too large.
+func durationToHAProxyTimespec(duration time.Duration) string {
+	haproxyMaxTimeoutMilliseconds := int64(2147483647)
+	durationMilliseconds := duration.Milliseconds()
+	if durationMilliseconds > haproxyMaxTimeoutMilliseconds {
+		log.V(7).Info("Time value %s exceeds the maximum timeout length of %dms. Truncating timeout to match maximum value.", duration.String(), haproxyMaxTimeoutMilliseconds)
```

Comment:
Should we log at level 2? It's only going to be "noisy" if this happens and, arguably, you'd like to find out quickly because we can't dynamically change the log level.

Reply:
I don't mind making it more verbose, but I chose `V(7)` to match the level of a very similar message generated by `clipHAProxyTimeoutValue`. Both seem like they're similar severity, and only could crop up when generating a new router deployment config if I understand correctly. Do you think that should also be reduced to level 2?

Reply:
We can deal with each separately. As as user I'd like to know if I hit some maximum. Logging at level 7 hides the clamping away.

Reply:
Fair enough. I'll make that change here.

Reply:
Done